src/capi/float.cpp

```diff
@@ -25,8 +27,45 @@ extern "C" {
 
 typedef enum { unknown_format, ieee_big_endian_format, ieee_little_endian_format } float_format_type;
 
+static float_format_type _detect_double_format() {
+    float_format_type format;
+#if SIZEOF_DOUBLE == 8
+    {
+        double x = 9006104071832581.0;
+        if (memcmp(&x, "\x43\x3f\xff\x01\x02\x03\x04\x05", 8) == 0)
+            format = ieee_big_endian_format;
+        else if (memcmp(&x, "\x05\x04\x03\x02\x01\xff\x3f\x43", 8) == 0)
+            format = ieee_little_endian_format;
+        else
+            format = unknown_format;
+    }
+#else
+    format = unknown_format;
+#endif
+    return format;
+}
+
+static float_format_type _detect_float_format() {
+    float_format_type format;
+#if SIZEOF_FLOAT == 4
+    {
+        float y = 16711938.0;
+        if (memcmp(&y, "\x4b\x7f\x01\x02", 4) == 0)
+            format = ieee_big_endian_format;
+        else if (memcmp(&y, "\x02\x01\x7f\x4b", 4) == 0)
+            format = ieee_little_endian_format;
+        else
+            format = unknown_format;
+    }
+#else
+    format = unknown_format;
+#endif
+    return format;
+}
+
 static float_format_type double_format, float_format;
-static float_format_type detected_double_format, detected_float_format;
+static float_format_type detected_double_format = _detect_double_format(),
+                         detected_float_format = _detect_float_format();
```

Comment:
oh doh @tjhance it looks like we should have added the float getformat stuff to this file (capi/float.cpp) rather than the one that we did in your pr (runtime/float.cpp).  it looks like we had added the variables here already and even have other code that looks at them, so I think it makes sense to have the implementation live here; mind if we move it over?